src/clj/ataru/tarjonta_service/tarjonta_service.clj

```diff
@@ -12,23 +12,28 @@
 (defn- any-hakuaika-on? [haku]
   (some true? (map #(hakuaika-on (:alkuPvm %) (:loppuPvm %)) (:hakuaikas haku))))
 
+(defn- haku-name-and-oid-when-hakuaika-on [accumulator haku]
+  (if (any-hakuaika-on? haku)
+    (assoc accumulator
+           (:oid haku)
+           {:haku-oid  (:oid haku)
+            :haku-name (get-in haku [:nimi :kieli_fi])})
+    accumulator))
+
+(defn- hakus-by-form-key [accumulator {:keys [avain haut]}]
+  (let [haku-info (reduce haku-name-and-oid-when-hakuaika-on {} haut)]
+    (if (not-empty haku-info)
+      (assoc accumulator avain haku-info)
```

Comment:
For sake of readability of code I would consider some other name than `accumulator` here. If the reduce calling this function returns set of hakus grouped by form keys I would consider something like `hakus` instead.

Reply:
Sure, i had a bad naming day